### PR TITLE
Add humidity from external sensor

### PR DESCRIPTION
**Pull request by LtHummus**
This PR adds the ability to read the humidity from an external sensor (namely PAC-USWHS003-TH-1). There should not be any additional configuration needed to make this work, though it will only work if direct access is enabled. 

I only was able to do hacky testing of this for those without temperature sensors, because all of my units have sensors attached to them.

### CODE REVIEW

**Review by fjs21 — CHANGES_REQUESTED** [review]
I know it is probably just window dressing but can you correct the spelling from 'Humdity' to 'Humidity'

**Comment by LtHummus** [comment]
Ha, good catch! (No one has ever accused me of being an accurate speller ;) ). I’ll fix that up tonight (and some of the things called out by the linter)

File: src/ductless.ts [hunk 1]
```diff
@@ -306,12 +333,40 @@ export class KumoPlatformAccessory_ductless {
...
+      return;
+    }
+    let currentValue: number = <number>this.Humdity.getCharacteristic(this.platform.Characteristic.CurrentRelativeHumidity).value;
+    if (this.accessory.context.sensors && this.accessory.context.sensors.length) {
+      const ourSensor = this.accessory.context.sensors[0];
+      currentValue = ourSensor.humidity;
+      this.platform.log.debug('setting humidity to %s', currentValue);
+
+      if (ourSensor.battery) {
+        this.Humdity.updateCharacteristic(this.platform.Characteristic.StatusLowBattery, ourSensor.battery < 10);
```
Comment (Jason-Morcos) [inline src/ductless.ts:362]: Would it be possible to add a battery level characteristic as well, in addition to the this.platform.Characteristic.StatusLowBattery characteristic? I believe there's one in the spec

**Review by Jason-Morcos — COMMENTED** [review]
This looks super cool, @LtHummus!

File: src/ductless.ts (hunk 1 above)
Comment (LtHummus) [reply src/ductless.ts:362]: I looked for one, but I only saw a low battery alert: https://developers.homebridge.io/#/service/HumiditySensor

Comment (Jason-Morcos) [reply src/ductless.ts:362]: I think it might be this? https://developers.homebridge.io/#/service/Battery

Comment (LtHummus) [reply src/ductless.ts:362]: Ahhh good call. I'll experiment with that tonight and see what I can make.

Comment (LtHummus) [reply src/ductless.ts]: Got this working...added a battery service and it seems to work as expected.

image
image

**Comment by LtHummus** [comment]
- Spelled `Humidity` correctly :)
- Fixed complaints from the linter (I usually write in Go at work, so you can see that seep in ;) )
- Added support to set the battery state

**Review by fjs21 — APPROVED** [review]

**Comment by Jason-Morcos** [comment]
Hi @fjs21! Do you have an estimated release date for a release that includes these changes? Is there a way to/would it be helpful to beta test this change in some way?

Thanks!

**Pull request merged**